### ISSUE TITLE
Deprecate the `missing` query in favour of a negated `exists` query

### ISSUE THREAD

**Issue by clintongormley** [elastic/elasticsearch#14112]
The `missing` query is implemented as a negation of the `exists` query, but it is trappy when used with nested documents: 

It can detect docs with nested docs which are missing a field, but cannot detect docs which have no nested docs. Instead, by using an exists filter inside the nested block and placing the negation outside it, things work as expected. (See #3495 for a full example)

To avoid this trap we should deprecate the `missing` filter in favour of using an `exists` filter where the user can place the negation where appropriate.

**Comment by jpountz** [comment]
+1

**Comment by jagdish-nasit** [comment]
+1

**Comment by jimferenczi** [comment]
The missing filter is able to filter (or not) null values (if the user specifies an explicit term for null values), this option is not present in the exists filter (it's in the documentation but not in the code ;) ). Should we add the feature in the exists filter before deprecation ?

**Comment by jpountz** [comment]
+1 to add it. If it's documented but not implemented, this sounds more like a bug to me.

**Comment by jimferenczi** [comment]
It's not documented sorry, the docs only explain the limitation regarding the filtering of null values ;)

**Comment by jpountz** [comment]
Actually after chatting with @jimferenczi, apparently the feature was never implemented. So maybe it's not worth working on it. Users can still search for `null` values explicitly by doing a `{"term": { "my_field": null }}`. @clintongormley any opinions?

**Comment by clintongormley** [comment]
The `missing` query supports `existence` and `null_value` parameters but honestly these are so hard to understand and explain that I don't think it is worth supporting these any more.  If a field has a `null_value` setting, then an explicit `null` will be replaced with a concrete value (and so will match an `exists` query).  If you want to exclude explicit nulls then you can add a clause which specifically excludes the term that you have used for the null value.  This I think is easier to understand than the `existence`/`null_value` parameters.

In other words: leave the `exists` query as it is today.

**Comment by jimferenczi** [comment]
Ok no problem, as a side note and as @clintongormley said: " If you want to exclude explicit nulls then you can add a clause which specifically excludes the term that you have used for the null value". Doing a {"term": { "my_field": null }} throws an exception ...
